### PR TITLE
persist forwards and calculate totals

### PR DESCRIPTION
**Pull request by JosephGoulden**

### CODE REVIEW

File: kld/src/cli/main.rs [hunk 1]
```diff
@@ -179,6 +179,14 @@ enum Command {
     },
     /// Fetch the aggregate local and remote channel balances (msat) of the node
     LocalRemoteBalance,
+    /// Get node routing fees.
+    GetFees,
+    /// Fetch a list of the forwarded htlcs.
+    ListForwards {
+        /// The status of the forwards (settled, failed)
+        #[arg(long)]
+        status: Option<String>,
```
Comment (yanganto) [inline kld/src/cli/main.rs:188]: Could we use `enum` for these? Or reuse the `ForwardStatus` in `kld/src/database/forward.rs`

```rust
enum Status {
  Settled,
  Failed
}

use std::str::FromStr;
type ParseError = &'static str;

impl FromStr for Status {
    type Err = ParseError;
    fn from_str(status: &str) -> Result<Self, Self::Err> {
        match status {
            "settled" => Ok(Status::Settled),
            "failed" => Ok(Status::Failed),
            _ => Err("Only status of the forwards allow now (settled,failed)"),
        }
    }
}
```

**Review by yanganto — CHANGES_REQUESTED** [review]

File: api/src/lib.rs [hunk 2]
```diff
@@ -37,25 +39,27 @@ pub mod routes {
...
+    pub const LIST_CHANNELS: &str = "/v1/channel/listPeerChannels";
     /// Open channel with a connected peer node.
     pub const OPEN_CHANNEL: &str = "/v1/channel/openChannel";
     /// Update channel fee policy.
     pub const SET_CHANNEL_FEE: &str = "/v1/channel/setChannelFee";
     /// Close an existing channel with a peer.
     pub const CLOSE_CHANNEL: &str = "/v1/channel/closeChannel/:id";
     /// Fetch aggregate channel local and remote balances.
-    pub const LOCAL_REMOTE_BALANCE: &str = "/v1/channel/localRemoteBal";
+    pub const LOCAL_REMOTE_BALANCE: &str = "/v1/channel/localremotebal";
```
[the remark this answers is not in the document]
Comment (JosephGoulden) [reply api/src/lib.rs:50]: Yeah that's a good idea. I'll do it in a separate PR as this one is big enough.

File: kld/src/cli/main.rs (hunk 1 above)
Comment (JosephGoulden) [reply kld/src/cli/main.rs:188]: The type it should be is GetV1ChannelListForwardsResponseItemStatus. But this is generated and doesn't seem to work well with clap due to missing traits. So I think we just have to leave as it is for now.

Comment (yanganto) [reply kld/src/cli/main.rs:188]: Just wondering, if the automatically generated code does not work, why not we write one and fix it?

Comment (JosephGoulden) [reply kld/src/cli/main.rs:188]: It's just less important to me than most other work we have to do. The API does the validation. CLI is a dumb client.

**Pull request merged**